assets/src/scripts/components/embed.js

```diff
@@ -0,0 +1,31 @@
+const { select } = require('d3-selection');
+
+
+/*
+ * Attaches a copy-to-clipboard function to a child input[type=text] node.
+ * @param {Object} store - Redux store
+ * @param {Object} node - DOM element
+ */
+function attachToNode(store, node) {
+    // Select all text in input when it gets focus
+    const input = select(node).select('input');
+    input.on('focus', function () {
+        this.setSelectionRange(0, this.value.length);
+    });
+
+    // Prepend a font awesome icon to the beginning of the node
+    select(node).insert('a', 'fieldset');
```

Comment:
Why not put this in the template?

Reply:
I was shooting for progressive behavior, where the input is visible when javascript is disabled.

Reply:
I must be missing something, but won't if the input is not visible then the icon would also be not visible?

Reply:
I added the class `no-script` to the markup, which will set it to visible when Javascript is disabled. There's also a `has-script` class for the opposite condition.

Reply:
Okay, I see what you are doing now.